src/core/diags/factory_diags.cpp

```diff
@@ -854,6 +872,12 @@ void Diags::TransmitDone(Error aError)
         }
         else
         {
+            if (!mRepeatActive && !mIsAsyncSend)
+            {
+                mIsAsyncSend = true;
+                Output("OT_ERROR_NONE");
```

Comment:
Do we want the output to say the literal `"OT_ERROR_NONE"` string here?

`AppendErrorResult()` seems to output the error as a hexadecimal value:

```cpp
void Diags::AppendErrorResult(Error aError)
{
    if ((aError != kErrorNone) && (aError != kErrorPending))
    {
        Output("failed\r\nstatus %#x\r\n", aError);
    }
}
```

Not related to this PR, but noticed that we are using `%#x` in the code here (which adds `0x` before the hexadecimal number). I suggest not using `%#` because it may not be supported on certain limited platforms/toolchains. Some simplified `printf` libraries on RTOS platforms provide limited format specifiers. It would be safer to avoid using it to make the code more portable.

Reply:
I have pushed a PR https://github.com/openthread/openthread/pull/11118 to remove all redundant messages from the diag module.